File: src/main/java/SpartJavaSample/Controllable.java
package SpartJavaSample;

import spark.Request;
import spark.Response;

import java.util.Map;

@SuppressWarnings("unused")
interface Controllable
{
    default boolean before(Request request, Response response) { return true; }

    default boolean get(Request request, Response response, final Map<String, Object> model) { return true; }

    default boolean post(Request request, Response response, final Map<String, Object> model) { return true; }

    default boolean put(Request request, Response response, final Map<String, Object> model) { return true; }

    default boolean delete(Request request, Response response, final Map<String, Object> model) { return true; }

    default boolean patch(Request request, Response response, final Map<String, Object> model) { return true; }

    default boolean head(Request request, Response response, final Map<String, Object> model) { return true; }

    default boolean trace(Request request, Response response, final Map<String, Object> model) { return true; }

    default boolean connect(Request request, Response response, final Map<String, Object> model) { return true; }

    default boolean options(Request request, Response response, final Map<String, Object> model) { return true; }

    default void after(Request request, Response response) { }
}

File: src/main/kotlin/SpartJavaSample/Controllable.kt
package SpartJavaSample

import spark.Request
import spark.Response

SuppressWarnings("unused")
abstract class Controllable {
    public open fun before(request: Request, response: Response): Boolean {
        return true
    }

    public open fun get(request: Request, response: Response, model: MutableMap<String, Any>): Boolean {
        return true
    }

    public open fun post(request: Request, response: Response, model: MutableMap<String, Any>): Boolean {
        return true
    }

    public open fun put(request: Request, response: Response, model: MutableMap<String, Any>): Boolean {
        return true
    }

    public open fun delete(request: Request, response: Response, model: MutableMap<String, Any>): Boolean {
        return true
    }

    public open fun patch(request: Request, response: Response, model: MutableMap<String, Any>): Boolean {
        return true
    }

    public open fun head(request: Request, response: Response, model: MutableMap<String, Any>): Boolean {
        return true
    }

    public open fun trace(request: Request, response: Response, model: MutableMap<String, Any>): Boolean {
        return true
    }

    public open fun connect(request: Request, response: Response, model: MutableMap<String, Any>): Boolean {
        return true
    }

    public open fun options(request: Request, response: Response, model: MutableMap<String, Any>): Boolean {
        return true
    }

    public open fun after(request: Request, response: Response) {
    }
}
